### ISSUE TITLE
Unable to load history page or history-graph

### ISSUE THREAD

**Issue by nottledim** [home-assistant/frontend#13721]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

History side panel doesn't load; there is just a continuous loading circle.

History-graph card for sensors displays "loading state history" indefinitely.  No graph is displayed.

### Describe the behavior you expected

I expect history panel to load the history
I expect history-graph to show a graph

### Steps to reproduce the issue

1. On home-assistant front-end select "History" from side panel menu
2.
3.
...

### What version of Home Assistant Core has the issue?

core-2022.9.2

### What was the last working version of Home Assistant Core?

core-2022.7.1

### In which browser are you experiencing the issue with?

Google chrome 104.0.5112.79 (Official Build) (64-bit), also firefox and HA app.

### Which operating system are you using to run this browser?

linux, android, chromeos etc.  HA server runs in docker on ArchLinux  5.17.9-arch1-1

### State of relevant entities

```txt
sensor.probe
Probe 21.2	
entity_id: sensor.probe_raw
unit_of_measurement: °C
icon: mdi:chart-line-variant
friendly_name: Probe

sensor.probe_raw
Probe Raw 21.5	
unit_of_measurement: °C
friendly_name: Probe Raw
```

### Problem-relevant frontend configuration

```yaml
- type: history-graph
    entities:
      - entity: climate.wiser_dick_2
      - entity: sensor.probe
    hours_to_show: 18
    title: Dick Room
```

### Javascript errors shown in your browser console/inspector

```txt
Panel:

Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'unit_of_measurement')
    at o (9f5922e0.js:1:5308)
    at d (9f5922e0.js:1:3481)
    at state-history-charts.ts:17:1
    at state-history-charts.ts:17:1
    at Array.forEach (<anonymous>)
    at h (state-history-charts.ts:17:1)
    at r.value (ha-panel-history.ts:115:1)
```
Card:
```
typography.js:181 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'substr')
    at i (typography.js:181:1)
    at n (e68243ce.js:1:665)
    at state-history-charts.ts:17:1
    at state-history-charts.ts:17:1
    at Array.map (<anonymous>)
    at h (state-history-charts.ts:17:1)
    at ha-circular-progress.ts:7:1
    at async a.value (hui-history-graph-card.ts:131:1)
```
I notice all errors related to this originate as uncaught promises "Cannot read properties of undefined". It doesn't elucidate the property it's expecting.
```


### Additional information

I use homeassistant docker image

I use mariadb for recording.   I have tried a virgin recorder using the standard sqlite data base (home-assistant_v2.db). Same issue.

History graphs of climate entities still work.  History graphs of sensor entities do not work.  My sensors are all derived from mqtt. Values are filtered to smooth them.  Hence probe_raw -> probe
Removing the filtering function does not resolve this issue.
Filter looks like this:
```
- platform: filter
  name: Probe
  entity_id: sensor.probe_raw
  filters:
    - filter: outlier
      window_size: 5
      radius: 0.5
    - filter: time_simple_moving_average
      window_size: "00:05"
      precision: 2
```

Equivalent History-Explorer card graphs using the same entities work without issue.

Issue home-assistant/frontend 13147 is relevant but was closed as fixed when it wasn't fixed for me.

Discussion here: https://community.home-assistant.io/t/history-not-working-after-latest-update/428368/7

**Comment by nottledim** [comment]
No, problem still exists on 2022.11.5.  There's been no response to my report.  I would like to get a resolution if possible.

**Comment by nottledim** [comment]
Problem still there 2022.12.4

**Comment by karwosts** [comment]
Is it possible to provide a complete set of instructions that could reproduce this in a fresh install? I tried making a copy of what you have in a dev environment (temperature sensor, and a filter probe of said temperature sensor), but I don't reproduce, both plot in the history with no problem. 

I think for someone to look at this more information would be needed. Not sure if there's more you can provide.

**Comment by nottledim** [comment]
I would if I knew what was interesting.   However I thought I might start a dev install from scratch and I notice the new system frontend is version "20221213.1 - latest" whereas my running system has "Frontend version: 20220203.0 - latest" and they look a bit different too.  Both systems are "2022.12.8"

How would I go about updating the frontend properly?

I would mention that at some point in the last year I replaced a git install of home-assistant with the docker container so some of the files in the config dir maybe unwanted cruft from original install.

**Comment by karwosts** [comment]
That's interesting. AFAIK there is no process to "update" the frontend, it is supposed to be packaged with the core and they update together. That's very odd to me if you've got such an old version with the latest core and would definitely explain some bugs you are seeing. 

The one way I do know to run with custom frontend is to have the `development_repo:` key in your configuration file under frontend, and in that case core runs with the frontend from a local folder, instead of with the one it was packaged with. Did you maybe set that at some point? 


https://www.home-assistant.io/integrations/frontend/
> development_repo string (Optional)
>
> Allows you to point to a directory containing frontend files instead of taking them from a pre-built PyPI package. Useful for Frontend development.

**Comment by nottledim** [comment]
no, never heard of that. I do, or did until a minute or, ago have an old install in the python directories (/usr/lib/python3.10/site-packages) a version of home assistant 2022.2.6 which is consistent with that old frontend version. Removing it has made no difference.

I've deleted .cache and deps in the config dir both of which contained files from Feb but no change. Reload browser cache and ...

Now things have changed. Frontend 20221213.1 - latest :-) And my graphs are back and history is working and I've got new menus. Hooray!

Thanks for taking an interest - it's just takes a nudge in a different direction. Much appreciated.
